src/CMakeLists.txt

```diff
@@ -11,7 +11,7 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 # Add additional CMake modules.
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-
+option(TINYSPLINE_WARNINGS_AS_ERRORS "Treat warnings as errors" ON)
```

Comment:
There is a dedicated options section: https://github.com/msteinbeck/tinyspline/blob/master/src/CMakeLists.txt#L280. Please put this statement into this section.